### PR TITLE
streamingccl: clean up stream ingestion job execution

### PR DESCRIPTION
**Pull request by msbutler**
This patch cleans up the stream ingestion job code by:
1. Seperating dist sql processor planning and execution in seperate functions.
2. Setting up the processor planning code for a future with job replanning.
3. Seperating stream ingestion and completion into seperat functions.
3. Cleaning up a few log lines and error messages.

Epic: none

Release note: none

### CODE REVIEW

**Comment by msbutler** [comment]
@lidorcarmel @stevendanna RFAL! CI failure is an unrelated test flake.

File: pkg/ccl/streamingccl/streamingest/stream_ingestion_dist.go [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+			return nil, nil, errors.Wrap(err, "failed to update job progress")
+		}
+
+		planCtx, sqlInstanceIDs, err := dsp.SetupAllNodesPlanning(ctx, execCtx.ExtendedEvalContext(), execCtx.ExecCfg())
+		if err != nil {
+			return nil, nil, err
+		}
+
+		// Construct stream ingestion processor specs.
+		streamIngestionSpecs, streamIngestionFrontierSpec, err := distStreamIngestionPlanSpecs(
```
Comment (stevendanna) [inline pkg/ccl/streamingccl/streamingest/stream_ingestion_dist.go:150]: [nit] If we feel like we need this comment, we should probably just name the function `constructStreamIngestionProcessorSpecs()`

File: pkg/ccl/streamingccl/streamingest/stream_ingestion_dist.go [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+		// Copy the evalCtx, as dsp.Run() might change it.
+		evalCtxCopy := *execCtx.ExtendedEvalContext()
+		dsp.Run(ctx, planCtx, noTxn, p, recv, &evalCtxCopy, nil /* finishedSetupFn */)
+		return rw.Err()
+	}
+
+	updateRunningStatus(ctx, execCtx, ingestionJob, jobspb.Replicating,
+		"running the SQL flow for the stream ingestion job")
+
+	// TODO(msbutler): implement automatic replanning in the spirit of changefeed replanning.
```
Comment (stevendanna) [inline pkg/ccl/streamingccl/streamingest/stream_ingestion_dist.go:111]: [nit, sorry] We usually try to use capitalised sentences for comments.

File: pkg/ccl/streamingccl/streamingest/stream_ingestion_dist.go [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+	client streamclient.Client,
+	previousHighWater hlc.Timestamp,
+	checkpoint jobspb.StreamIngestionCheckpoint,
+	initialScanTimestamp hlc.Timestamp,
+) func(context.Context, *sql.DistSQLPlanner) (*sql.PhysicalPlan, *sql.PlanningCtx, error) {
+	jobID := ingestionJob.ID()
+	return func(ctx context.Context, dsp *sql.DistSQLPlanner) (*sql.PhysicalPlan, *sql.PlanningCtx, error) {
+		log.Infof(ctx, "Re Planning DistSQL flow for stream ingestion job %d", jobID)
+
+		streamID := streampb.StreamID(details.StreamID)
```
Comment (stevendanna) [inline pkg/ccl/streamingccl/streamingest/stream_ingestion_dist.go:130]: It doesn't matter, but if you wanted, this and gatewayID below could live outside the returned function along with jobID. Or jobID could live in here.

**Review by stevendanna — COMMENTED** [review]
Left a few nitpicks, but otherwise this cleanup seems fine to me. We should backport it to make other backports easier.

**Review by msbutler — COMMENTED** [review]
@stevendanna ready for a quick look

**Review by stevendanna — APPROVED** [review]
Looks reasonable to me.  Note that this is one step in GA-relevant work around replanning so I think we should be OK to backport it.

**Comment by msbutler** [comment]
blathers backport 23.1

**Pull request merged**